cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1191,57 +1194,6 @@ set(Boost_USE_MULTITHREADED ON)
 if(MSVC AND ARROW_USE_STATIC_CRT)
   set(Boost_USE_STATIC_RUNTIME ON)
 endif()
-set(Boost_ADDITIONAL_VERSIONS
```

Comment:
Do we not need to keep this around for anyone running an older cmake version + boost without the config? (our min is 3.16 after all)

Reply:
Let's try our CI.
If we need to keep this, it tells us. :-)

Reply:
Hmm.
Our vcpkg doesn't have `boost-cmake` yet...: #43812